### PR TITLE
Improve error message when lazy collection is not initialized

### LINKED ISSUE #1205: Improve error message when lazy collection is not initialized
Currently, the error is:
```
org.hibernate.LazyInitializationException: HR000056: Collection cannot be initialized: com.example.Parent.lazyChildren
```

We should tell the user that they need to fetch the connection:
* Using a `fetch join` in the query
* Or, using the `Mutiny.fetch` or the `Stage.fetch`

We should also add this on the WebSite in the Faq section.

### PR DESCRIPTION
**Pull request by blafond**
fixes #1205

### CODE REVIEW

File: hibernate-reactive-core/src/main/java/org/hibernate/reactive/logging/impl/Log.java [hunk 1]
```diff
@@ -170,7 +170,7 @@ public interface Log extends BasicLogger {
 	@Message(id = 55, value = "Session closed")
 	LazyInitializationException sessionClosedLazyInitializationException();
 
-	@Message(id = 56, value = "Collection cannot be initialized: %1$s")
+	@Message(id = 56, value = "Collection cannot be initialized: %1$s\nFetch the connection by using Mutiny.fetch, Stage.fetch or use `fetch join` in your query")
```
[the remark this answers is not in the document]
Comment (gavinking) [reply hibernate-reactive-core/src/main/java/org/hibernate/reactive/logging/impl/Log.java:173]: But I would definitely avoid backticks there. Oh and "collection", not "connection".

How about:

    fetch the collection using 'Mutiny.fetch', 'Stage.fetch' or 'fetch join' in HQL

[the remark this answers is not in the document]
Comment (gavinking) [reply hibernate-reactive-core/src/main/java/org/hibernate/reactive/logging/impl/Log.java]: Excellent thanks 🙏

**Review by DavideD — CHANGES_REQUESTED** [review]
Thanks!
The error message looks good but I don't think we need the additional test cases for fetching associations. We already test fetching for the different type of associations elsewhere and these classes test  that we throw the right error message when a lazy initialization error occurs.

I would prefer to keep them simpler. Unless you think that a specific use case isn't covered by our test suit.

**Comment by blafond** [comment]
Looking through our tests, we don't have many that have entities with uninitialized `List<..>` or `Set<>` and they don't test this use-case.

There's only a 2 that use `Stage.fetch()`  (collection is initialized) and none that use `Mutiny.fetch()`

These new tests cover all scenarios in one class.

**Comment by DavideD** [comment]
> There's only a 2 that use Stage.fetch() (collection is initialized) and none that use Mutiny.fetch() 

? 

I haven't checked each single use case but these are all the classes  testing Mutiny.fetch:

* LazyOrderedElementCollectionForEmbeddableTypeListTest
* ManyToManyMapTest
* ManyToManySetTest
* ManyToManyTest
* NonNullableManyToOneTest
* OneToManyMapTest
* OneToManySetTest
* OneToManyTest
* OrderedEmbeddableCollectionTest
* OrderedManyToManyTest
* OrderedOneToManyTestMutiny
* SubselectElementCollectionForEmbeddableTypeListTest

And these are the test classes for `Stage.fetch`:

* BatchFetchTest
* CascadeComplicatedTest
* CascadeComplicatedToOnesEagerTest
* CascadeTest
* EagerUniqueKeyTest
* FilterTest
* LazyManyToOneAssociationTest
* LazyOneToManyAssociationWithFetchTest
* LazyUniqueKeyTest
* OrphanRemovalTest
* SubselectFetchTest

What's different between the additional tests and the ones in 
`LazyOneToManyAssociationWithFetchTest` and `OneToManyTest`?

Also, the additional tests don't need to be in the same commit you are using to update the log message. It would be nice to have them in a separate  commit (they are not related to this fix).

**Comment by blafond** [comment]
Most of the tests that use fetch for collections have entities that are initialized in the entity definition (i.e. new ArrayList<...> , new HashSet<...>) or define instances of the collection via custom constructor etc.

If it's important to have a test that has a true NULL array definition and the collection is initialized internally then I think adding these tests with versions of `fetch` methodology makes sense.

**Pull request merged**